src/test/scala/edu/berkeley/ce/rockslicing/EndToEndSpec.scala

```diff
@@ -34,57 +34,36 @@ class EndToEndSpec extends FunSuite {
       Block(center, block.nonRedundantFaces)
     }
 
-    // Note: The following two filter operations could be done using the partition method;
-    //       however, when executing on Spark this will be an RDD and partition will not work.
-    //       The two filter operations are left here to mimic what is done in the actual Spark code
-
-    // Find all blocks that contain processor joints
-    val processorBlocks = nonRedundantBlocks.filter { block =>
-      block.faces.exists(_.processorJoint)
-    }
-
     // Find blocks that do not contain processor joints
-    val realBlocks = nonRedundantBlocks.filter { block =>
-      val faceTests = block.faces.exists(_.processorJoint)
-      !faceTests
-    }
-
-    // Search blocks for matching processor joints
-    val updatedProcessorBlocks = processorBlocks.map { block =>
-      Block(globalOrigin, block.updateFaces(globalOrigin))
+    val (processorBlocks, realBlocks) = nonRedundantBlocks.partition { block => block.faces.exists(_.isProcessorFace) }
+    val finalBlocks = if (processorBlocks.isEmpty) {
+      realBlocks
+    } else {
+      realBlocks ++ LoadBalancer.mergeProcessorBlocks(processorBlocks)
```

Comment:
I guess `mergeProcessorBlocks` only really gets used in the unit tests since the code had to be adapted to deal with `RDD`'s?